### PR TITLE
Splash screen containing the loading bar

### PR DESCRIPTION
**Pull request by filipeotero**
This PR contains the design implementation for the dynamic splash as described on the splash screen Wiki page:
https://wiki.autodesk.com/display/GEN/Dynamo+splash+screen

For the implementation of the grid, I have installed the bootstrap package that makes the stylization more simple.
https://react-bootstrap.github.io/layout/grid/

Here is an example of how it works by inputting some data by calling a method from outside the web app.
Also, for this example, I created a sandbox WPF containing a popup to simulate how it's going to work in Dynamo.

dynamicSplashScreen

FYI: @QilongTang @reddyashish @zeusongit @RobertGlobant20 @jesusalvino

### CODE REVIEW

File: package.json
Comment (zeusongit) [inline package.json:15]: Do we really need each one of these dependencies? I was thinking since we don't have a complex UI use case we could implement this using bare-minimum external dependencies. Totally fine if it is not possible, but just a thought to analyze each dependency individually as it will also improve future maintainability.

File: src/Dynamic.js
Comment (zeusongit) [inline src/Dynamic.js:14]: I suppose this will be accessed from Dynamo?

File: src/App.js
Comment (zeusongit) [inline src/App.js:55]: Any specific reason to use images in b64 formats? Maybe to update from inside Dynamo?

File: package.json
Comment (mjkkirschner) [reply package.json:15]: 👍 - we have to keep all of these dependencies updated as vulnerabilities are discovered, as well as keep the license info up to date.

we can follow the pattern here:
https://github.com/DynamoDS/NotificationCenter/pull/25 (thats fine in another PR).

Comment (mjkkirschner) [reply package.json:15]: is web-vitals a peer dep of something? I don't see it used.
https://web.dev/vitals/

Comment (filipeotero) [reply package.json]: Hi guys, thanks for noticing that. I removed the unused packages and I guess we can update the license in another PR.

File: src/Dynamic.js
Comment (filipeotero) [reply src/Dynamic.js:14]: Yes, that's right.

**Comment by Amoursol** [comment]
@filipeotero we'll need to change that image out as that is a Revit image :) Let me take a look to see if we have a better one from Dynamo itself.

**Comment by jesusalvino** [comment]
Any chance to load the image from a shared place like a S3 bucket so it could be changed it easily ?

**Comment by filipeotero** [comment]
> Any chance to load the image from a shared place like a S3 bucket so it could be changed it easily ?

That's a good idea. What do you think? @zeusongit

**Comment by QilongTang** [comment]
> > Any chance to load the image from a shared place like a S3 bucket so it could be changed it easily ?
> 
> That's a good idea. What do you think? @zeusongit

I actually think we do not want to do that. That would means when user is offline, they would not see the image at all from our splash screen which is not a common experience for other ADSK products. Since the splash screen is the first thing user see in Dynamo, I do not want to make any comprimise yet in similar user case even offline.

File: .gitignore
Comment (QilongTang) [inline .gitignore:24]: seems even this is included, the content from dist folder is still included in this PR? Please remove

File: src/App.js
Comment (QilongTang) [reply src/App.js:55]: Can we put this embeded image in a different js file, I am worried this big block will affect local dev and PR review in the future

File: package.json
Comment (QilongTang) [inline package.json:7]: what is this used for?

File: src/App.js
Comment (QilongTang) [inline src/App.js:27]: Can we have a more genric class name, e.g. WelcomeRow etc

File: src/Dynamic.js
Comment (QilongTang) [reply src/Dynamic.js:14]: This is great. Some more comments in this file for each function would be appreciated

**Review by QilongTang — COMMENTED** [review]
Good progres, some comments

**Comment by QilongTang** [comment]
@filipeotero I would like to also consult you with the font, is it embedded currently in the css file? What is the font?

File: package.json
Comment (filipeotero) [reply package.json]: I thought it was used for the bootstrap, but I tested and it's working fine without it. Removed.

File: .gitignore
Comment (filipeotero) [reply .gitignore]: added a '/' and now its ignoring the files

**Review by QilongTang — COMMENTED** [review]
LGTM once rest of the comments addressed and also the image from @Amoursol updated

File: .gitignore
Comment (QilongTang) [reply .gitignore]: @filipeotero Thanks!

File: src/App.js
Comment (QilongTang) [inline src/App.js:36]: Can you make this settable from Dynamo but with this default value? I ask only because we currently have a blocker on the localization side, so we may need to use Dynamo to pass in localized string for this one. The other resources are fine, e.g. the title `Dynamo` does not need to be localized

**Comment by filipeotero** [comment]
> > > Any chance to load the image from a shared place like a S3 bucket so it could be changed it easily ?
> > 
> > 
> > That's a good idea. What do you think? @zeusongit
> 
> I actually think we do not want to do that. That would means when user is offline, they would not see the image at all from our splash screen which is not a common experience for other ADSK products. Since the splash screen is the first thing user see in Dynamo, I do not want to make any comprimise yet in similar user case even offline.

@QilongTang I created a method called `setBackgroundImage` with a keyword "#base64BackgroundImage" that can be replaced when we read the files on the Dynamo side.

File: src/App.js
Comment (filipeotero) [reply src/App.js]: Done, Created a new js file called `encodedImages` containing the long strings

**Comment by QilongTang** [comment]
> > > > Any chance to load the image from a shared place like a S3 bucket so it could be changed it easily ?
> > > 
> > > 
> > > That's a good idea. What do you think? @zeusongit
> > 
> > 
> > I actually think we do not want to do that. That would means when user is offline, they would not see the image at all from our splash screen which is not a common experience for other ADSK products. Since the splash screen is the first thing user see in Dynamo, I do not want to make any comprimise yet in similar user case even offline.
> 
> @QilongTang I created a method called `setBackgroundImage` with a keyword "#base64BackgroundImage" that can be replaced when we read the files on the Dynamo side.

Thanks, this should give us more flexibility

**Review by QilongTang — APPROVED** [review]

**Comment by QilongTang** [comment]
I think this is good as a first version to get in. @filipeotero Let me know if you plan to actually replaced the encoded image in this repo before merging

**Comment by filipeotero** [comment]
> I think this is good as a first version to get in. @filipeotero Let me know if you plan to actually replaced the encoded image in this repo before merging

Image replaced:

image (2)

**Pull request merged**
